includes/html/api_functions.inc.php

```diff
@@ -2598,6 +2599,23 @@ function del_service_from_host(Illuminate\Http\Request $request)
     return api_error(500, 'Failed to delete the service');
 }
 
+function search_by_mac(Illuminate\Http\Request $request)
+{
+    $search = $request->route('search');
+    $portlist = PortsFdb::select('port_id')
```

Comment:
I think this should return all ports matching the Mac Instead?
Something like
```
Port::whereHas('fdbEntries', function($q) {
   $q->where('mac_address', $search);
})->get();
```

Reply:
I just want to get the Downlink.
e.g.
```
PortsFdb::whereIn('port_id',PortsFdb::where('mac_address', $search)->select('port_id'))->groupBy('port_id')->orderByRaw('count(port_id)')->select('port_id')->first()->port;
```

Reply:
@mpikzink your code has two queries, but the inside on isn't actually executed. Also, it looks like you want to select the port, so perhaps you should start with Port:: for the query.

Reply:
@murrat Yes I need a port. But I am looking for a port that "only" has this MAC in the FDB. Traditionally as SQL query I can only achieve this via a subquery so that I can count the remaining MAC addresses on the port in the FDB.

Reply:
Remember that a MAC can be on many ports (switch downlinks).

Subquery is supported, but that is not the correct syntax

Reply:
I know that it can be on many ports but I think if I search for a client the client will be on the port with the fewest count of other MACs. I have this (first commit) option in our setup since a year and it works very well for our client search which we use multiple times per day. I will have a look for the right syntax for laravel. 

BTW: Since my first commit I think my chosen route is also not the best place and I need to describe in the doc as well.